### PR TITLE
Shaokai/modelzoo quadruped

### PR DESCRIPTION
**Pull request by yeshaokai**
Top-down quadruped video inference code using a third party package. 

NOTE, this branch should not be deleted, as it in part supports results in Ye et al, 2023 SuperAnimal work.

### CODE REVIEW

**Comment by MMathisLab** [comment]
@yeshaokai please run code formatting, black, clean up spelling, and any new files that are not part of truly third_party please add DLC header.

File: deeplabcut/superanimal_pytorch/image_inference.sh
Comment (MMathisLab) [inline deeplabcut/superanimal_pytorch/image_inference.sh:1]: why is this file needed? Its also hard-coded for @yeshaokai

File: deeplabcut/superanimal_pytorch/third_party/demo/top_down_video_demo_with_mmdet.py
Comment (MMathisLab) [inline deeplabcut/superanimal_pytorch/third_party/demo/top_down_video_demo_with_mmdet.py:1]: needed? seems to ref. human pose est.

File: deeplabcut/superanimal_pytorch/third_party/det_config_SA_quadruped.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        rcnn=dict(
+            score_thr=0.05, nms=dict(type="nms", iou_threshold=0.5), max_per_img=100
+        )
+        # soft-nms is also supported for rcnn testing
+        # e.g.,
+    ),
+)
+# dataset settings
+dataset_type = "DLCGenericDataset"
+data_root = "data/ma_superquadruped_coco/"
```
Comment (MMathisLab) [inline deeplabcut/superanimal_pytorch/third_party/det_config_SA_quadruped.py:124]: issue with hard coding?

File: deeplabcut/superanimal_pytorch/third_party/mmpose/ViT_models/pose_tokenpose_b.py
Comment (MMathisLab) [inline deeplabcut/superanimal_pytorch/third_party/mmpose/ViT_models/pose_tokenpose_b.py:1]: all of these models are not mmpose, so should be moved to a separate "Microsoft" third_party folder and code updated as needed.

File: deeplabcut/superanimal_pytorch/third_party/mmpose/core/post_processing/nms.py
Comment (MMathisLab) [inline deeplabcut/superanimal_pytorch/third_party/mmpose/core/post_processing/nms.py:1]: move to Microsoft third_party folder

**Review by MMathisLab — CHANGES_REQUESTED** [review]
changes needed:

- refactoring of subfolders; i.e. those that are not core mmpose, such as all the microsoft code, should go into a separate third_party folder. 

-please add DLC headers to any new DLC files

- please check for hard-coded paths and remove any run files not needed

- please perform a stress-test with @n-poulsen and @jeylau and Anastasiia before merging.

File: deeplabcut/superanimal_pytorch/image_inference.sh
Comment (yeshaokai) [reply deeplabcut/superanimal_pytorch/image_inference.sh:1]: This is for steffen to use for web app. Paths are references that are to be changed in the backend server.

File: deeplabcut/superanimal_pytorch/third_party/demo/top_down_video_demo_with_mmdet.py
Comment (yeshaokai) [reply deeplabcut/superanimal_pytorch/third_party/demo/top_down_video_demo_with_mmdet.py:1]: Yes. This is the inference code.

Comment (MMathisLab) [reply deeplabcut/superanimal_pytorch/third_party/demo/top_down_video_demo_with_mmdet.py:1]: then can you modify comments to make it clearer :) add adapted by at header

File: deeplabcut/superanimal_pytorch/image_inference.sh
Comment (MMathisLab) [reply deeplabcut/superanimal_pytorch/image_inference.sh:1]: I think should go to the openmodelzoo repo then, not dlc main

File: deeplabcut/superanimal_pytorch/third_party/det_config_SA_quadruped.py (hunk 1 above)
Comment (yeshaokai) [reply deeplabcut/superanimal_pytorch/third_party/det_config_SA_quadruped.py:124]: Not a problem. Those hardcoded paths are only used at training time. At inference time, those variables are not used (but needed to be defined otherwise it throws errors)

File: deeplabcut/superanimal_pytorch/image_inference.sh
Comment (yeshaokai) [reply deeplabcut/superanimal_pytorch/image_inference.sh:1]: Deleted. Steffen can use the one in commit history as his draft for webapp

File: deeplabcut/superanimal_pytorch/third_party/mmpose/ViT_models/pose_tokenpose_b.py
Comment (yeshaokai) [reply deeplabcut/superanimal_pytorch/third_party/mmpose/ViT_models/pose_tokenpose_b.py:1]: After I talked to Nasya, I did further clean up and this ViT_models folder is removed. But do we really want to move all files that have microsoft header to microsoft folder? Doing this will break the registry pattern and it becomes hard to keep track of things.

**Comment by MMathisLab** [comment]
Closing in favor of new pytorch DLC version, which is forthcoming 🔥
